manager/assets/modext/widgets/core/modx.panel.js

```diff
@@ -313,56 +369,111 @@ Ext.extend(MODx.FormPanel,Ext.FormPanel,{
-     * @param {String} tabsId - id of tab component for a given panel
+     * @param {Object} map - the items.map object of the primary tabs panel
+     * @param {Array} keys - an array of keys matching those in the tabsObj items.keys
      */
-    ,showErroredTab: function(detectingForms, tabsId) {
-        var tab = null, index = null;
-        for (var i = 0; i < detectingForms.length; i++) {
-            var component = Ext.getCmp(detectingForms[i]);
-            if (component && component.el && component.el.dom) {
-                if (this.detectErrors(component.el.dom)) {
-                    tab = component.itemId ? component.itemId : detectingForms[i];
-                    break;
+    ,getTabIdsFromKeys: function(map, keys) {
+
+        let tabIds = [];
+
+        if (typeof map == 'object') {
+            if (Array.isArray(keys) && keys.length > 0) {
+                keys.forEach(function(key) {
+                    if(map.hasOwnProperty(key) && typeof map[key].id == 'string') {
+                        tabIds.push(map[key].id);
+                    } else if (key == 'modx-panel-resource-tv' && MODx.config.tvs_below_content == 1) {
+                        /*
+                            When evaluating a resource panel with TVs moved below content,
+                            the panel id needs to be added explicitly as, in this case, the TV panel
+                            is not part of the main tabs component
+                        */
+                        tabIds.push(key);
                 }
+                });
             }
         }
-
-        if (tab === null) {
-            return;
+        return tabIds;
         }
 
-        var tabs = Ext.getCmp(tabsId);
-
-        if (tabs && tabs.items && tabs.items.keys) {
-            index = tabs.items.keys.indexOf(tab);
+    /**
+     * @property {Function} showErroredTab - Find errored field in the panel and activates the tab where the first error was found.
+     *
+     * @param {Array} targetForms - array of form tab itemIds to search for errors
+     * @param {String} tabsId - id of primary tab component for a given panel
+     */
+    ,showErroredTab: function(targetForms, tabsId) {
+
+        const mainTabs = Ext.getCmp(tabsId);
+        let searchTabs = this.getTabIdsFromKeys(mainTabs.items.map, targetForms),
+            mainTabName = null,
+            mainTabIndex = null,
+            component,
+            erroredNode = null
+        ;
+        /*
+            Add any custom panels, created on the fly via manager customization or CMPs,
+            to the searchTabs
+        */
+        if (mainTabs.items.length > mainTabs.initialConfig.items.length) {
+            mainTabs.items.keys.forEach(function(key) {
+                if (mainTabs.items.map[key].hasOwnProperty('id')) {
+                    if(this.errorHandlingIgnoreTabs.indexOf(mainTabs.items.map[key].id) === -1 && searchTabs.indexOf(mainTabs.items.map[key].id) === -1) {
```

Comment:
There is no need to know the index of the element.
```suggestion
                    if (!this.errorHandlingIgnoreTabs.includes(mainTabs.items.map[key].id) && !searchTabs.includes(mainTabs.items.map[key].id)) {
```